pkg/controller/v1beta1/inferenceservice/controller.go

```diff
@@ -296,26 +296,33 @@ func inferenceServiceStatusEqual(s1, s2 v1beta1api.InferenceServiceStatus, deplo
 }
 
 func (r *InferenceServiceReconciler) SetupWithManager(mgr ctrl.Manager, deployConfig *v1beta1api.DeployConfig, ingressConfig *v1beta1api.IngressConfig) error {
-	if deployConfig.DefaultDeploymentMode == string(constants.RawDeployment) {
-		return ctrl.NewControllerManagedBy(mgr).
-			For(&v1beta1api.InferenceService{}).
-			Owns(&appsv1.Deployment{}).
-			Complete(r)
-	} else if ingressConfig.DisableIstioVirtualHost == false {
-		return ctrl.NewControllerManagedBy(mgr).
-			For(&v1beta1api.InferenceService{}).
-			Owns(&knservingv1.Service{}).
-			Owns(&istioclientv1beta1.VirtualService{}).
-			Owns(&appsv1.Deployment{}).
-			Complete(r)
+	ksvcFound, err := utils.IsCrdAvailable(mgr.GetConfig(), "serving.knative.dev/v1", "Service")
+	if err != nil {
+		return err
+	}
+
+	vsFound, err := utils.IsCrdAvailable(mgr.GetConfig(), "networking.istio.io/v1beta1", "VirtualService")
+	if err != nil {
+		return err
+	}
+
+	ctrlBuilder := ctrl.NewControllerManagedBy(mgr).
+		For(&v1beta1api.InferenceService{}).
+		Owns(&appsv1.Deployment{})
+
+	if ksvcFound {
+		ctrlBuilder = ctrlBuilder.Owns(&knservingv1.Service{})
+	} else {
+		r.Log.Info("The InferenceService controller won't watch serving.knative.dev/v1/Service resources because the CRD is not available.")
+	}
+
+	if vsFound && ingressConfig.DisableIstioVirtualHost == false {
+		ctrlBuilder = ctrlBuilder.Owns(&istioclientv1beta1.VirtualService{})
 	} else {
-		return ctrl.NewControllerManagedBy(mgr).
-			For(&v1beta1api.InferenceService{}).
-			Owns(&knservingv1.Service{}).
-			Owns(&appsv1.Deployment{}).
-			Complete(r)
+		r.Log.Info("The InferenceService controller won't watch networking.istio.io/v1beta1/VirtualService resources because the CRD is not available.")
 	}
 
+	return ctrlBuilder.Complete(r)
```

Comment:
Are you planning to add tests to check if the configuration is correctly applied after the reconcile runs?

Reply:
What configuration are you referring to?

Reply:
I was referring to the isvc, e.g.
if you change the isvc and force knative, and if it is disabled, the isvc should not be reconciled, right? so the confs should be the same than before running the reconcile loop again.

does it make sense?

Reply:
At some point in time, I added to this PR some units in `pkg/controller/v1beta1/inferenceservice/controller_test.go` and `pkg/controller/v1alpha1/inferencegraph/controller_test.go`. Maybe that's enough?